src/main/features/core/remote/index.ts

```diff
@@ -0,0 +1,569 @@
+import { Stats, promises } from 'fs';
+import { readFile } from 'fs/promises';
+import { IncomingMessage, Server, ServerResponse, createServer } from 'http';
+import { join } from 'path';
+import { deflate, gzip } from 'zlib';
+import axios from 'axios';
+import { app, ipcMain } from 'electron';
+import { Server as WsServer, WebSocketServer, WebSocket } from 'ws';
+import { ClientEvent, ServerEvent } from '../../../../remote/types';
+import { PlayerRepeat, SongUpdate } from '../../../../renderer/types';
+import { getMainWindow } from '../../../main';
+import { isLinux } from '../../../utils';
+import { store } from '../settings/index';
+
+let mprisPlayer: any | undefined;
+
+if (isLinux()) {
+    // eslint-disable-next-line global-require
+    mprisPlayer = require('../../linux/mpris').mprisPlayer;
+}
+
+interface RemoteConfig {
+    enabled: boolean;
+    port: number;
+}
+
+interface MimeType {
+    css: string;
+    html: string;
+    ico: string;
+    js: string;
+}
+
+interface StatefulWebSocket extends WebSocket {
+    alive: boolean;
+}
+
+let server: Server | undefined;
+let wsServer: WsServer<StatefulWebSocket> | undefined;
+
+function broadcast({ event, data }: ServerEvent): void {
+    if (wsServer) {
+        for (const client of wsServer.clients) {
+            if (client.readyState === WebSocket.OPEN) {
+                client.send(JSON.stringify({ data, event }));
+            }
+        }
+    }
+}
+
+type SendData = ServerEvent & {
+    client: StatefulWebSocket;
+};
+
+function send({ client, event, data }: SendData): void {
+    if (client.readyState === WebSocket.OPEN) {
+        client.send(JSON.stringify({ data, event }));
+    }
+}
```

Comment:
For `broadcast()` why not incorporate `send()` inside since it's got the same logic as broadcast?

```suggestion
type SendData = ServerEvent & {
    client: StatefulWebSocket;
};

function send({ client, event, data }: SendData): void {
    if (client.readyState === WebSocket.OPEN) {
        client.send(JSON.stringify({ data, event }));
    }
}

function broadcast({ event, data }: ServerEvent): void {
    if (wsServer) {
        for (const client of wsServer.clients) {
          send({ client, event, data });
        }
    }
}
```